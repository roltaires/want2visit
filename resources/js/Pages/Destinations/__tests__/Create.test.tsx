import React from "react";
import { render, unmountComponentAtNode } from "react-dom";
import { fireEvent } from '@testing-library/react'
import pretty from "pretty";

import Create from "@/Pages/Destinations/Create";

let container : HTMLDivElement;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement("div");
  document.body.appendChild(container);
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
});

const auth = {
  user: {
    name: "Test",
    email: "test@example.com",
  }
}

it("renders properly", () => { 
  render(<Create auth={auth} errors={{}} />, container);
});

it("displays all form inputs", () => {
  render(<Create auth={auth} errors={{}} />, container);

  expect(container.querySelectorAll(`
    [data-testid='create-form'] input,
    [data-testid='create-form'] select,
    [data-testid='create-form'] textarea,
    [data-testid='create-form'] button`).length).toEqual(5);
})

it("can update input values", () => {
  render(<Create auth={auth} errors={{}} />, container);

  let locationInput : HTMLInputElement = container.querySelector('[data-testid="create-form"] [name="location"]')!; 
  fireEvent.input(locationInput, {target: {value: "Davao, Philippines"}});
  expect(locationInput.value).toEqual('Davao, Philippines');

  let date_month_select : HTMLInputElement = container.querySelector('[data-testid="create-form"] [name="date_month"]')!; 
  fireEvent.select(date_month_select, {target: {value: 4}});
  const dateMonthVal = date_month_select.value;
  expect(container.querySelector(`[data-testid="create-form"] [name="date_month"] option[value="${dateMonthVal}"]`)?.textContent).toEqual('May');

  let date_year_select : HTMLInputElement = container.querySelector('[data-testid="create-form"] [name="date_year"]')!; 
  fireEvent.select(date_year_select, {target: {value: 2024}});
  const dateYearVal = date_year_select.value;
  expect(container.querySelector(`[data-testid="create-form"] [name="date_year"] option[value="${dateYearVal}"]`)?.textContent).toEqual('2024');

  let reasonsInput : HTMLInputElement = container.querySelector('[data-testid="create-form"] [name="reasons"]')!; 
  fireEvent.input(reasonsInput, {target: {value: "I\'m told it\'s safe there?"}});
  expect(reasonsInput.value).toEqual('I\'m told it\'s safe there?');

  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
})

it("can submit new destination", async () => {
  render(<Create auth={auth} errors={{}} />, container);

  let locationInput : HTMLInputElement = container.querySelector('[data-testid="create-form"] [name="location"]')!; 
  fireEvent.input(locationInput, {target: {value: "Davao, Philippines"}});
  expect(locationInput.value).toEqual('Davao, Philippines');

  let date_month_select : HTMLInputElement = container.querySelector('[data-testid="create-form"] [name="date_month"]')!; 
  fireEvent.select(date_month_select, {target: {value: 4}});
  const dateMonthVal = date_month_select.value;
  expect(container.querySelector(`[data-testid="create-form"] [name="date_month"] option[value="${dateMonthVal}"]`)?.textContent).toEqual('May');

  let date_year_select : HTMLInputElement = container.querySelector('[data-testid="create-form"] [name="date_year"]')!; 
  fireEvent.select(date_year_select, {target: {value: 2024}});
  const dateYearVal = date_year_select.value;
  expect(container.querySelector(`[data-testid="create-form"] [name="date_year"] option[value="${dateYearVal}"]`)?.textContent).toEqual('2024');

  let reasonsInput : HTMLInputElement = container.querySelector('[data-testid="create-form"] [name="reasons"]')!; 
  fireEvent.input(reasonsInput, {target: {value: "I\'m told it\'s safe there?"}});
  expect(reasonsInput.value).toEqual('I\'m told it\'s safe there?');

  fireEvent.submit(container.querySelector('[data-testid="create-form"]')!);
  
  expect(
    pretty(container.innerHTML)
  ).toMatchSnapshot();
})