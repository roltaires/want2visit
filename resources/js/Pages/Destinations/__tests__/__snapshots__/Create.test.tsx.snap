// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`can submit new destination 1`] = `
"<div class=\\"min-h-screen bg-gray-100\\">
  <nav class=\\"bg-white border-b border-gray-100\\">
    <div class=\\"max-w-7xl mx-auto px-4 sm:px-6 lg:px-8\\">
      <div class=\\"flex justify-between h-16\\">
        <div class=\\"flex\\">
          <div class=\\"shrink-0 flex items-center\\"><a href=\\"/\\"><svg class=\\"block h-9 w-auto text-gray-500\\" viewBox=\\"0 0 316 316\\" xmlns=\\"http://www.w3.org/2000/svg\\">
                <path d=\\"M305.8 81.125C305.77 80.995 305.69 80.885 305.65 80.755C305.56 80.525 305.49 80.285 305.37 80.075C305.29 79.935 305.17 79.815 305.07 79.685C304.94 79.515 304.83 79.325 304.68 79.175C304.55 79.045 304.39 78.955 304.25 78.845C304.09 78.715 303.95 78.575 303.77 78.475L251.32 48.275C249.97 47.495 248.31 47.495 246.96 48.275L194.51 78.475C194.33 78.575 194.19 78.725 194.03 78.845C193.89 78.955 193.73 79.045 193.6 79.175C193.45 79.325 193.34 79.515 193.21 79.685C193.11 79.815 192.99 79.935 192.91 80.075C192.79 80.285 192.71 80.525 192.63 80.755C192.58 80.875 192.51 80.995 192.48 81.125C192.38 81.495 192.33 81.875 192.33 82.265V139.625L148.62 164.795V52.575C148.62 52.185 148.57 51.805 148.47 51.435C148.44 51.305 148.36 51.195 148.32 51.065C148.23 50.835 148.16 50.595 148.04 50.385C147.96 50.245 147.84 50.125 147.74 49.995C147.61 49.825 147.5 49.635 147.35 49.485C147.22 49.355 147.06 49.265 146.92 49.155C146.76 49.025 146.62 48.885 146.44 48.785L93.99 18.585C92.64 17.805 90.98 17.805 89.63 18.585L37.18 48.785C37 48.885 36.86 49.035 36.7 49.155C36.56 49.265 36.4 49.355 36.27 49.485C36.12 49.635 36.01 49.825 35.88 49.995C35.78 50.125 35.66 50.245 35.58 50.385C35.46 50.595 35.38 50.835 35.3 51.065C35.25 51.185 35.18 51.305 35.15 51.435C35.05 51.805 35 52.185 35 52.575V232.235C35 233.795 35.84 235.245 37.19 236.025L142.1 296.425C142.33 296.555 142.58 296.635 142.82 296.725C142.93 296.765 143.04 296.835 143.16 296.865C143.53 296.965 143.9 297.015 144.28 297.015C144.66 297.015 145.03 296.965 145.4 296.865C145.5 296.835 145.59 296.775 145.69 296.745C145.95 296.655 146.21 296.565 146.45 296.435L251.36 236.035C252.72 235.255 253.55 233.815 253.55 232.245V174.885L303.81 145.945C305.17 145.165 306 143.725 306 142.155V82.265C305.95 81.875 305.89 81.495 305.8 81.125ZM144.2 227.205L100.57 202.515L146.39 176.135L196.66 147.195L240.33 172.335L208.29 190.625L144.2 227.205ZM244.75 114.995V164.795L226.39 154.225L201.03 139.625V89.825L219.39 100.395L244.75 114.995ZM249.12 57.105L292.81 82.265L249.12 107.425L205.43 82.265L249.12 57.105ZM114.49 184.425L96.13 194.995V85.305L121.49 70.705L139.85 60.135V169.815L114.49 184.425ZM91.76 27.425L135.45 52.585L91.76 77.745L48.07 52.585L91.76 27.425ZM43.67 60.135L62.03 70.705L87.39 85.305V202.545V202.555V202.565C87.39 202.735 87.44 202.895 87.46 203.055C87.49 203.265 87.49 203.485 87.55 203.695V203.705C87.6 203.875 87.69 204.035 87.76 204.195C87.84 204.375 87.89 204.575 87.99 204.745C87.99 204.745 87.99 204.755 88 204.755C88.09 204.905 88.22 205.035 88.33 205.175C88.45 205.335 88.55 205.495 88.69 205.635L88.7 205.645C88.82 205.765 88.98 205.855 89.12 205.965C89.28 206.085 89.42 206.225 89.59 206.325C89.6 206.325 89.6 206.325 89.61 206.335C89.62 206.335 89.62 206.345 89.63 206.345L139.87 234.775V285.065L43.67 229.705V60.135ZM244.75 229.705L148.58 285.075V234.775L219.8 194.115L244.75 179.875V229.705ZM297.2 139.625L253.49 164.795V114.995L278.85 100.395L297.21 89.825V139.625H297.2Z\\"></path>
              </svg></a></div>
          <div class=\\"hidden space-x-8 sm:-my-px sm:ml-10 sm:flex\\"><a class=\\"inline-flex items-center px-1 pt-1 border-b-2 border-transparent text-sm font-medium leading-5 text-gray-500 hover:text-gray-700 hover:border-gray-300 focus:outline-none focus:text-gray-700 focus:border-gray-300 transition duration-150 ease-in-out\\" href=\\"[object%20Object]\\">Dashboard</a><a class=\\"inline-flex items-center px-1 pt-1 border-b-2 border-transparent text-sm font-medium leading-5 text-gray-500 hover:text-gray-700 hover:border-gray-300 focus:outline-none focus:text-gray-700 focus:border-gray-300 transition duration-150 ease-in-out\\" href=\\"[object%20Object]\\">Destinations</a></div>
        </div>
        <div class=\\"hidden sm:flex sm:items-center sm:ml-6\\">
          <div class=\\"ml-3 relative\\">
            <div class=\\"relative\\">
              <div><span class=\\"inline-flex rounded-md\\"><button type=\\"button\\" class=\\"inline-flex items-center px-3 py-2 border border-transparent text-sm leading-4 font-medium rounded-md text-gray-500 bg-white hover:text-gray-700 focus:outline-none transition ease-in-out duration-150\\">Test<svg class=\\"ml-2 -mr-0.5 h-4 w-4\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\"><path fill-rule=\\"evenodd\\" d=\\"M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z\\" clip-rule=\\"evenodd\\"></path></svg></button></span></div>
            </div>
          </div>
        </div>
        <div class=\\"-mr-2 flex items-center sm:hidden\\"><button class=\\"inline-flex items-center justify-center p-2 rounded-md text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 focus:text-gray-500 transition duration-150 ease-in-out\\"><svg class=\\"h-6 w-6\\" stroke=\\"currentColor\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\">
              <path class=\\"inline-flex\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M4 6h16M4 12h16M4 18h16\\"></path>
              <path class=\\"hidden\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M6 18L18 6M6 6l12 12\\"></path>
            </svg></button></div>
      </div>
    </div>
    <div class=\\"hidden sm:hidden\\">
      <div class=\\"pt-2 pb-3 space-y-1\\"><a class=\\"w-full flex items-start pl-3 pr-4 py-2 border-l-4 border-transparent text-gray-600 hover:text-gray-800 hover:bg-gray-50 hover:border-gray-300 text-base font-medium focus:outline-none transition duration-150 ease-in-out\\" href=\\"[object%20Object]\\">Dashboard</a></div>
      <div class=\\"pt-4 pb-1 border-t border-gray-200\\">
        <div class=\\"px-4\\">
          <div class=\\"font-medium text-base text-gray-800\\">Test</div>
          <div class=\\"font-medium text-sm text-gray-500\\">test@example.com</div>
        </div>
        <div class=\\"mt-3 space-y-1\\"><button class=\\"w-full flex items-start pl-3 pr-4 py-2 border-l-4 border-transparent text-gray-600 hover:text-gray-800 hover:bg-gray-50 hover:border-gray-300 text-base font-medium focus:outline-none transition duration-150 ease-in-out\\">Log Out</button></div>
      </div>
    </div>
  </nav>
  <header class=\\"bg-white shadow\\">
    <div class=\\"max-w-7xl mx-auto py-6 px-4 sm:px-6 lg:px-8\\">
      <h2 class=\\"font-semibold text-xl text-gray-800 leading-tight\\">Create Destination</h2>
    </div>
  </header>
  <main>
    <div class=\\"py-12\\">
      <div class=\\"max-w-7xl mx-auto sm:px-6 lg:px-8\\">
        <div class=\\"bg-white overflow-hidden shadow-sm sm:rounded-lg\\">
          <div class=\\"p-6 bg-white border-b border-gray-200\\">
            <form data-testid=\\"create-form\\">
              <div class=\\"mb-3\\"><label for=\\"location\\">Location</label>
                <div class=\\"flex flex-col items-start\\"><input type=\\"text\\" name=\\"location\\" id=\\"location\\" class=\\"border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm w-full\\" value=\\"Davao, Philippines\\"></div>
              </div>
              <div class=\\"mb-3\\"><label for=\\"date\\">When do I plan to go there?</label>
                <div><select name=\\"date_month\\" class=\\"border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm mr-2\\">
                    <option value=\\"3\\">April</option>
                    <option value=\\"4\\">May</option>
                    <option value=\\"5\\">June</option>
                    <option value=\\"6\\">July</option>
                    <option value=\\"7\\">August</option>
                    <option value=\\"8\\">September</option>
                    <option value=\\"9\\">October</option>
                    <option value=\\"10\\">November</option>
                    <option value=\\"11\\">December</option>
                  </select><select name=\\"date_year\\" class=\\"border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm undefined\\">
                    <option value=\\"2022\\">2022</option>
                    <option value=\\"2023\\">2023</option>
                    <option value=\\"2024\\">2024</option>
                    <option value=\\"2025\\">2025</option>
                    <option value=\\"2026\\">2026</option>
                    <option value=\\"2027\\">2027</option>
                    <option value=\\"2028\\">2028</option>
                    <option value=\\"2029\\">2029</option>
                    <option value=\\"2030\\">2030</option>
                    <option value=\\"2031\\">2031</option>
                    <option value=\\"2032\\">2032</option>
                    <option value=\\"2033\\">2033</option>
                    <option value=\\"2034\\">2034</option>
                    <option value=\\"2035\\">2035</option>
                    <option value=\\"2036\\">2036</option>
                    <option value=\\"2037\\">2037</option>
                    <option value=\\"2038\\">2038</option>
                    <option value=\\"2039\\">2039</option>
                    <option value=\\"2040\\">2040</option>
                    <option value=\\"2041\\">2041</option>
                    <option value=\\"2042\\">2042</option>
                    <option value=\\"2043\\">2043</option>
                    <option value=\\"2044\\">2044</option>
                    <option value=\\"2045\\">2045</option>
                    <option value=\\"2046\\">2046</option>
                    <option value=\\"2047\\">2047</option>
                    <option value=\\"2048\\">2048</option>
                    <option value=\\"2049\\">2049</option>
                    <option value=\\"2050\\">2050</option>
                    <option value=\\"2051\\">2051</option>
                    <option value=\\"2052\\">2052</option>
                    <option value=\\"2053\\">2053</option>
                    <option value=\\"2054\\">2054</option>
                    <option value=\\"2055\\">2055</option>
                    <option value=\\"2056\\">2056</option>
                    <option value=\\"2057\\">2057</option>
                    <option value=\\"2058\\">2058</option>
                    <option value=\\"2059\\">2059</option>
                    <option value=\\"2060\\">2060</option>
                    <option value=\\"2061\\">2061</option>
                    <option value=\\"2062\\">2062</option>
                    <option value=\\"2063\\">2063</option>
                    <option value=\\"2064\\">2064</option>
                    <option value=\\"2065\\">2065</option>
                    <option value=\\"2066\\">2066</option>
                    <option value=\\"2067\\">2067</option>
                    <option value=\\"2068\\">2068</option>
                    <option value=\\"2069\\">2069</option>
                    <option value=\\"2070\\">2070</option>
                    <option value=\\"2071\\">2071</option>
                    <option value=\\"2072\\">2072</option>
                  </select></div>
              </div>
              <div class=\\"mb-3\\"><label for=\\"reasons\\">Why do I want to go there?</label>
                <div class=\\"flex flex-col items-start\\"><textarea name=\\"reasons\\" id=\\"reasons\\" class=\\"border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm w-full\\" rows=\\"3\\">I'm told it's safe there?</textarea></div>
              </div><button type=\\"submit\\" class=\\"inline-flex items-center px-4 py-2 bg-gray-800 border border-transparent rounded-md font-semibold text-xs text-white uppercase tracking-widest active:bg-gray-900 transition ease-in-out duration-150 opacity-25 \\" disabled=\\"\\">Submit</button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>"
`;

exports[`can update input values 1`] = `
"<div class=\\"min-h-screen bg-gray-100\\">
  <nav class=\\"bg-white border-b border-gray-100\\">
    <div class=\\"max-w-7xl mx-auto px-4 sm:px-6 lg:px-8\\">
      <div class=\\"flex justify-between h-16\\">
        <div class=\\"flex\\">
          <div class=\\"shrink-0 flex items-center\\"><a href=\\"/\\"><svg class=\\"block h-9 w-auto text-gray-500\\" viewBox=\\"0 0 316 316\\" xmlns=\\"http://www.w3.org/2000/svg\\">
                <path d=\\"M305.8 81.125C305.77 80.995 305.69 80.885 305.65 80.755C305.56 80.525 305.49 80.285 305.37 80.075C305.29 79.935 305.17 79.815 305.07 79.685C304.94 79.515 304.83 79.325 304.68 79.175C304.55 79.045 304.39 78.955 304.25 78.845C304.09 78.715 303.95 78.575 303.77 78.475L251.32 48.275C249.97 47.495 248.31 47.495 246.96 48.275L194.51 78.475C194.33 78.575 194.19 78.725 194.03 78.845C193.89 78.955 193.73 79.045 193.6 79.175C193.45 79.325 193.34 79.515 193.21 79.685C193.11 79.815 192.99 79.935 192.91 80.075C192.79 80.285 192.71 80.525 192.63 80.755C192.58 80.875 192.51 80.995 192.48 81.125C192.38 81.495 192.33 81.875 192.33 82.265V139.625L148.62 164.795V52.575C148.62 52.185 148.57 51.805 148.47 51.435C148.44 51.305 148.36 51.195 148.32 51.065C148.23 50.835 148.16 50.595 148.04 50.385C147.96 50.245 147.84 50.125 147.74 49.995C147.61 49.825 147.5 49.635 147.35 49.485C147.22 49.355 147.06 49.265 146.92 49.155C146.76 49.025 146.62 48.885 146.44 48.785L93.99 18.585C92.64 17.805 90.98 17.805 89.63 18.585L37.18 48.785C37 48.885 36.86 49.035 36.7 49.155C36.56 49.265 36.4 49.355 36.27 49.485C36.12 49.635 36.01 49.825 35.88 49.995C35.78 50.125 35.66 50.245 35.58 50.385C35.46 50.595 35.38 50.835 35.3 51.065C35.25 51.185 35.18 51.305 35.15 51.435C35.05 51.805 35 52.185 35 52.575V232.235C35 233.795 35.84 235.245 37.19 236.025L142.1 296.425C142.33 296.555 142.58 296.635 142.82 296.725C142.93 296.765 143.04 296.835 143.16 296.865C143.53 296.965 143.9 297.015 144.28 297.015C144.66 297.015 145.03 296.965 145.4 296.865C145.5 296.835 145.59 296.775 145.69 296.745C145.95 296.655 146.21 296.565 146.45 296.435L251.36 236.035C252.72 235.255 253.55 233.815 253.55 232.245V174.885L303.81 145.945C305.17 145.165 306 143.725 306 142.155V82.265C305.95 81.875 305.89 81.495 305.8 81.125ZM144.2 227.205L100.57 202.515L146.39 176.135L196.66 147.195L240.33 172.335L208.29 190.625L144.2 227.205ZM244.75 114.995V164.795L226.39 154.225L201.03 139.625V89.825L219.39 100.395L244.75 114.995ZM249.12 57.105L292.81 82.265L249.12 107.425L205.43 82.265L249.12 57.105ZM114.49 184.425L96.13 194.995V85.305L121.49 70.705L139.85 60.135V169.815L114.49 184.425ZM91.76 27.425L135.45 52.585L91.76 77.745L48.07 52.585L91.76 27.425ZM43.67 60.135L62.03 70.705L87.39 85.305V202.545V202.555V202.565C87.39 202.735 87.44 202.895 87.46 203.055C87.49 203.265 87.49 203.485 87.55 203.695V203.705C87.6 203.875 87.69 204.035 87.76 204.195C87.84 204.375 87.89 204.575 87.99 204.745C87.99 204.745 87.99 204.755 88 204.755C88.09 204.905 88.22 205.035 88.33 205.175C88.45 205.335 88.55 205.495 88.69 205.635L88.7 205.645C88.82 205.765 88.98 205.855 89.12 205.965C89.28 206.085 89.42 206.225 89.59 206.325C89.6 206.325 89.6 206.325 89.61 206.335C89.62 206.335 89.62 206.345 89.63 206.345L139.87 234.775V285.065L43.67 229.705V60.135ZM244.75 229.705L148.58 285.075V234.775L219.8 194.115L244.75 179.875V229.705ZM297.2 139.625L253.49 164.795V114.995L278.85 100.395L297.21 89.825V139.625H297.2Z\\"></path>
              </svg></a></div>
          <div class=\\"hidden space-x-8 sm:-my-px sm:ml-10 sm:flex\\"><a class=\\"inline-flex items-center px-1 pt-1 border-b-2 border-transparent text-sm font-medium leading-5 text-gray-500 hover:text-gray-700 hover:border-gray-300 focus:outline-none focus:text-gray-700 focus:border-gray-300 transition duration-150 ease-in-out\\" href=\\"[object%20Object]\\">Dashboard</a><a class=\\"inline-flex items-center px-1 pt-1 border-b-2 border-transparent text-sm font-medium leading-5 text-gray-500 hover:text-gray-700 hover:border-gray-300 focus:outline-none focus:text-gray-700 focus:border-gray-300 transition duration-150 ease-in-out\\" href=\\"[object%20Object]\\">Destinations</a></div>
        </div>
        <div class=\\"hidden sm:flex sm:items-center sm:ml-6\\">
          <div class=\\"ml-3 relative\\">
            <div class=\\"relative\\">
              <div><span class=\\"inline-flex rounded-md\\"><button type=\\"button\\" class=\\"inline-flex items-center px-3 py-2 border border-transparent text-sm leading-4 font-medium rounded-md text-gray-500 bg-white hover:text-gray-700 focus:outline-none transition ease-in-out duration-150\\">Test<svg class=\\"ml-2 -mr-0.5 h-4 w-4\\" xmlns=\\"http://www.w3.org/2000/svg\\" viewBox=\\"0 0 20 20\\" fill=\\"currentColor\\"><path fill-rule=\\"evenodd\\" d=\\"M5.293 7.293a1 1 0 011.414 0L10 10.586l3.293-3.293a1 1 0 111.414 1.414l-4 4a1 1 0 01-1.414 0l-4-4a1 1 0 010-1.414z\\" clip-rule=\\"evenodd\\"></path></svg></button></span></div>
            </div>
          </div>
        </div>
        <div class=\\"-mr-2 flex items-center sm:hidden\\"><button class=\\"inline-flex items-center justify-center p-2 rounded-md text-gray-400 hover:text-gray-500 hover:bg-gray-100 focus:outline-none focus:bg-gray-100 focus:text-gray-500 transition duration-150 ease-in-out\\"><svg class=\\"h-6 w-6\\" stroke=\\"currentColor\\" fill=\\"none\\" viewBox=\\"0 0 24 24\\">
              <path class=\\"inline-flex\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M4 6h16M4 12h16M4 18h16\\"></path>
              <path class=\\"hidden\\" stroke-linecap=\\"round\\" stroke-linejoin=\\"round\\" stroke-width=\\"2\\" d=\\"M6 18L18 6M6 6l12 12\\"></path>
            </svg></button></div>
      </div>
    </div>
    <div class=\\"hidden sm:hidden\\">
      <div class=\\"pt-2 pb-3 space-y-1\\"><a class=\\"w-full flex items-start pl-3 pr-4 py-2 border-l-4 border-transparent text-gray-600 hover:text-gray-800 hover:bg-gray-50 hover:border-gray-300 text-base font-medium focus:outline-none transition duration-150 ease-in-out\\" href=\\"[object%20Object]\\">Dashboard</a></div>
      <div class=\\"pt-4 pb-1 border-t border-gray-200\\">
        <div class=\\"px-4\\">
          <div class=\\"font-medium text-base text-gray-800\\">Test</div>
          <div class=\\"font-medium text-sm text-gray-500\\">test@example.com</div>
        </div>
        <div class=\\"mt-3 space-y-1\\"><button class=\\"w-full flex items-start pl-3 pr-4 py-2 border-l-4 border-transparent text-gray-600 hover:text-gray-800 hover:bg-gray-50 hover:border-gray-300 text-base font-medium focus:outline-none transition duration-150 ease-in-out\\">Log Out</button></div>
      </div>
    </div>
  </nav>
  <header class=\\"bg-white shadow\\">
    <div class=\\"max-w-7xl mx-auto py-6 px-4 sm:px-6 lg:px-8\\">
      <h2 class=\\"font-semibold text-xl text-gray-800 leading-tight\\">Create Destination</h2>
    </div>
  </header>
  <main>
    <div class=\\"py-12\\">
      <div class=\\"max-w-7xl mx-auto sm:px-6 lg:px-8\\">
        <div class=\\"bg-white overflow-hidden shadow-sm sm:rounded-lg\\">
          <div class=\\"p-6 bg-white border-b border-gray-200\\">
            <form data-testid=\\"create-form\\">
              <div class=\\"mb-3\\"><label for=\\"location\\">Location</label>
                <div class=\\"flex flex-col items-start\\"><input type=\\"text\\" name=\\"location\\" id=\\"location\\" class=\\"border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm w-full\\" value=\\"Davao, Philippines\\"></div>
              </div>
              <div class=\\"mb-3\\"><label for=\\"date\\">When do I plan to go there?</label>
                <div><select name=\\"date_month\\" class=\\"border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm mr-2\\">
                    <option value=\\"3\\">April</option>
                    <option value=\\"4\\">May</option>
                    <option value=\\"5\\">June</option>
                    <option value=\\"6\\">July</option>
                    <option value=\\"7\\">August</option>
                    <option value=\\"8\\">September</option>
                    <option value=\\"9\\">October</option>
                    <option value=\\"10\\">November</option>
                    <option value=\\"11\\">December</option>
                  </select><select name=\\"date_year\\" class=\\"border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm undefined\\">
                    <option value=\\"2022\\">2022</option>
                    <option value=\\"2023\\">2023</option>
                    <option value=\\"2024\\">2024</option>
                    <option value=\\"2025\\">2025</option>
                    <option value=\\"2026\\">2026</option>
                    <option value=\\"2027\\">2027</option>
                    <option value=\\"2028\\">2028</option>
                    <option value=\\"2029\\">2029</option>
                    <option value=\\"2030\\">2030</option>
                    <option value=\\"2031\\">2031</option>
                    <option value=\\"2032\\">2032</option>
                    <option value=\\"2033\\">2033</option>
                    <option value=\\"2034\\">2034</option>
                    <option value=\\"2035\\">2035</option>
                    <option value=\\"2036\\">2036</option>
                    <option value=\\"2037\\">2037</option>
                    <option value=\\"2038\\">2038</option>
                    <option value=\\"2039\\">2039</option>
                    <option value=\\"2040\\">2040</option>
                    <option value=\\"2041\\">2041</option>
                    <option value=\\"2042\\">2042</option>
                    <option value=\\"2043\\">2043</option>
                    <option value=\\"2044\\">2044</option>
                    <option value=\\"2045\\">2045</option>
                    <option value=\\"2046\\">2046</option>
                    <option value=\\"2047\\">2047</option>
                    <option value=\\"2048\\">2048</option>
                    <option value=\\"2049\\">2049</option>
                    <option value=\\"2050\\">2050</option>
                    <option value=\\"2051\\">2051</option>
                    <option value=\\"2052\\">2052</option>
                    <option value=\\"2053\\">2053</option>
                    <option value=\\"2054\\">2054</option>
                    <option value=\\"2055\\">2055</option>
                    <option value=\\"2056\\">2056</option>
                    <option value=\\"2057\\">2057</option>
                    <option value=\\"2058\\">2058</option>
                    <option value=\\"2059\\">2059</option>
                    <option value=\\"2060\\">2060</option>
                    <option value=\\"2061\\">2061</option>
                    <option value=\\"2062\\">2062</option>
                    <option value=\\"2063\\">2063</option>
                    <option value=\\"2064\\">2064</option>
                    <option value=\\"2065\\">2065</option>
                    <option value=\\"2066\\">2066</option>
                    <option value=\\"2067\\">2067</option>
                    <option value=\\"2068\\">2068</option>
                    <option value=\\"2069\\">2069</option>
                    <option value=\\"2070\\">2070</option>
                    <option value=\\"2071\\">2071</option>
                    <option value=\\"2072\\">2072</option>
                  </select></div>
              </div>
              <div class=\\"mb-3\\"><label for=\\"reasons\\">Why do I want to go there?</label>
                <div class=\\"flex flex-col items-start\\"><textarea name=\\"reasons\\" id=\\"reasons\\" class=\\"border-gray-300 focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 rounded-md shadow-sm w-full\\" rows=\\"3\\">I'm told it's safe there?</textarea></div>
              </div><button type=\\"submit\\" class=\\"inline-flex items-center px-4 py-2 bg-gray-800 border border-transparent rounded-md font-semibold text-xs text-white uppercase tracking-widest active:bg-gray-900 transition ease-in-out duration-150 false \\">Submit</button>
            </form>
          </div>
        </div>
      </div>
    </div>
  </main>
</div>"
`;
